### PR TITLE
Unify and beef up random IR generator

### PR DESCRIPTION
**Pull request by meheff**
Create a single random IR generator which is used everywhere such as for fuzz tests. The generator covers many more corner cases. Also fix new bugs found along the way.

Also, bump to v0.53.0.

### CODE REVIEW

File: xlsynth-g8r/fuzz/src/lib.rs [hunk 1]
```diff
@@ -57,3 +62,103 @@ pub fn build_graph(sample: &FuzzGraph) -> Option<GateFn> {
...
+    generate_upstream_formal_random_pir_package(data, package_name)
+}
+
+/// Generates two upstream-compatible functions with matching signatures for
+/// toolchain-backed equivalence properties.
+pub fn generate_upstream_formal_random_pir_pair(data: &[u8]) -> (IrFn, IrFn) {
+    let options = upstream_formal_random_pir_options(64);
+    let mut entropy = DepletableBytes::new(data);
+    let (first, second) =
+        generate_same_signature_pair(&mut entropy, &options, StopPolicy::WhenEntropyDepleted)
```
[the remark this answers is not in the document]
Comment (meheff) [reply xlsynth-g8r/fuzz/src/lib.rs]: fixed.
